File: ArrayAndHashing/hasDuplicate.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        //first sort the list
        int temp;
        for(int i = 0; i < nums.size(); i++){
            for(int j = i + 1; j < nums.size(); j++){
                if(nums[i] > nums[j]){
                    temp = nums[i];
                    nums[i] = nums[j];
                    nums[j] = temp;
                }
            }
        }

        //check if there are any duplicates
        for(int v = 0; v < nums.size(); v++){
            if(nums[v] == nums[v+1]){
                return true;
            } 
        }
        return false;
        
    }
};

File: ArrayAndHashing/hasDuplicate.py
class Solution:
    #Time: O(n^2), Space: O(1)
    def hasDuplicate(self, nums: List[int]) -> bool:
        #check each element with another and see if it matches
         for n in range(len(nums)):
            for j in range(n+1, len(nums)):
                if nums[n] == nums[j]:
                    return True
         return False

#ANOTHER WAY OF DOING THIS SIMILAR TO isAnagram: Time: O(n) Space: O(n)
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        #init hashmap
         hashmap = {}

        #loop through the list and add each occurence of the number (the key will be the # in nums, and value would be # of occurences)
         for i in range(len(nums)):
            hashmap[nums[i]] = 1 + hashmap.get(nums[i], 0)

        #loop through the hashmap and see if there is a number occurring more than once
         for n in hashmap:
            if hashmap[n] > 1:
                return True
        
         return False
